### DISCUSSION TITLE
Custom Named Versioning

### DISCUSSION THREAD

**Discussion by wparr-circle** [renovatebot/renovate#27496] [Request Help]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

37.186.1

### If you're self-hosting Renovate, select which platform you are using.

github.com

### What is your question?

Hi,

We would like to be able to create custom named versioning, to prevent duplication of similar regex patterns and make different variants available for use where the existing library of supported versions won't work for us: https://docs.renovatebot.com/modules/versioning/#supported-versioning

We had thought of using `versioningTemplate` here - but not quite sure how to structure a template using [handlebars builtin lookup helper](https://handlebarsjs.com/guide/builtin-helpers.html#lookup) given we don't have access to handlebars helpers such as [jsonparse](https://github.com/helpers/handlebars-helpers?tab=readme-ov-file#JSONparse).

We did manage to get this functioning using if/else, but that would only support a single named versioning as opposed to multiple.

With:
```json
"versioningTemplate": "{{#if (equals versioning 'compatibility-suffix')}}regex:^v?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?-(?<compatibility>.*)${{else}}{{{versioning}}}{{/if}}"
```
Setting versioning to `versioning=compatibility-suffix`

Which results in the following versioning be used after applying the template:
```json
"versioning":"regex:^v?(?<major>\\d+)\\.(?<minor>\\d+)\\.(?<patch>\\d+)?-(?<compatibility>.*)$"
```

As an aside, I was wondering if rather than attempting to do this via templating - should renovate support a `customVersioning` concept, similar to how we can define:
- [custom datasource](https://docs.renovatebot.com/modules/datasource/custom/)
- [custom managers](https://docs.renovatebot.com/modules/manager/regex/)
etc.

### Logs (if relevant)

<details><summary>Logs</summary>

```

Copy/paste the relevant log(s) here, between the starting and ending backticks

```

</details>

**Comment by rarkins** [comment]
Can this be done via `extends`?

e.g. you create a preset which contains only a `versioning` definition

Then you could extend it from other configs
